v2/client/invoker.go

```diff
@@ -118,12 +118,18 @@ func (r *receiveInvoker) IsResponder() bool {
 	return r.fn.hasEventOut
 }
 
-func extractContextFromMessage(message binding.Message, fallback context.Context) context.Context {
+// extractContextFromMessage attempts to extract a context.Context from message. If message does not implement
+// MessageContext, or MessageContext.Context() returns nil, then fallback is returned.
+func extractContextFromMessage(message binding.Message, fallback context.Context) (ctx context.Context) {
 	if mctx, ok := message.(binding.MessageContext); ok {
-		return mctx.Context()
+		ctx = mctx.Context()
 	} else if mctx, ok := binding.UnwrapMessage(message).(binding.MessageContext); ok {
-		return mctx.Context()
-	} else {
+		ctx = mctx.Context()
+	}
+
+	if ctx == nil {
 		return fallback
 	}
+
+	return ctx
```

Comment:
changes to this function means that if `MessageContext.Context()` returns `nil`, then `fallback` is returned, whereas previously it would return `nil`. 

Not sure if we can do this as it may be a breaking change, however I'd say this is safer since using a `nil` context is typically a bad idea anyway.

Reply:
I'm ok with this change, thus as you outlined before, using another context is a breaking change